app/decorators/feature-flag-route.ts

```diff
@@ -0,0 +1,55 @@
+import { service } from '@ember-decorators/service';
+import { assert } from '@ember/debug';
+import Route from '@ember/routing/route';
+import { Registry as Services } from '@ember/service';
+import config from 'ember-get-config';
+import RSVP from 'rsvp';
+
+import transitionTarget from 'ember-osf-web/utils/transition-target';
+
+const {
+    featureFlags: {
+        routes,
+    },
+} = config as { featureFlags: { routes: { [index: string]: string } } }; // eslint-disable-line no-use-before-define
+
+/**
+ * Class decorator for feature flagging Routes.
+ * In the `beforeModel` hook, check whether the route is feature flagged.
+ * If the route is not flagged, proceed normally.
+ * If the route is flagged and the route feature is enabled, proceed normally.
+ * If the route is flagged and the route feature is not enabled, redirect to route url (forcing a page load).
+ *
+ * Use:
+ * ```
+ * // Check if route is feature flag and force page load if it's flagged, but not enabled.
+ * @featureFlagRoute()
+ * class MyRoute extends Route { ... }
+ *
+ * Uses the mixin pattern here: https://github.com/Microsoft/TypeScript/pull/13743
+ */
+export default function featureFlagRouteFactory() {
+    function featureFlagRouteDecorator<T extends Newable<Route>>(RouteSubclass: T) {
+        class FeatureFlaggedRoute extends RouteSubclass {
+            @service features!: Services['features'];
+            @service router!: Services['router'];
+
+            beforeModel(transition: any) {
+                const flag = routes[transition.targetName];
```

Comment:
How would you feel about passing the flag name into the decorator, like
```ts
@featureFlagRoute('ember_dashboard_page')
export default class Dashboard extends Route {
```
and removing the route feature flags from the config? It seems cleaner, but I don't know if we expect to use different flags by environment.

Reply:
I was also thinking about this, so +1 to the question. As far as different configs, I think having different flag names on different environments would be a bad idea.

Reply:
Was planning to add this as discussed, but this *would* use the same flag for multiple engine mounts (which has been ok, so far....)